src/nvim/terminal.c

```diff
@@ -217,10 +214,14 @@ Terminal *terminal_open(TerminalOptions opts)
   vterm_set_utf8(rv->vt, 1);
   // Setup state
   VTermState *state = vterm_obtain_state(rv->vt);
+  // Only relevant in terminal true color mode
+  if (true_color) {
+    vterm_state_set_bold_highbright(state, true);
```

Comment:
Oh, right. In order to not regress #3333, let's add a `g:terminal_bold_highbright` setting. 

Though your suggestion at https://github.com/neovim/neovim/pull/4274#issuecomment-185291835 avoids a new setting, it will require more code and more documentation to explain it. 

Avoiding another `g:terminal_` option right now doesn't buy us much until all of the `g:terminal` options are replaced with some other scheme, so, adding a new one makes no difference.